helper/configure-image.sh

```diff
@@ -24,26 +27,34 @@ else
     exit 1
 fi
 
-# Install package manager dependencies
+# Install dependencies using the determined package manager
 case "$PACKAGE_MANAGER" in
-        apt-get)
-            apt-get update
-            apt-get install -qq curl
-            ;;
-        apk)
-            apk add --update-cache --upgrade --virtual .temp-dde-deps curl shadow
+    apt-get)
+        apt-get update
+        apt-get install -qq curl
+        [ "$SHELL_TO_INSTALL" = "/bin/zsh" ] && apt-get install -y zsh
+        ;;
+    apk)
+        apk add --update-cache --upgrade --virtual .temp-dde-deps curl shadow
+        [ "$SHELL_TO_INSTALL" = "/bin/zsh" ] && apk add zsh
+        ;;
 esac
 
-# Install gosu
+# Install gosu if not already installed
 if [ ! -x /usr/local/bin/gosu ]; then
     curl -L https://github.com/tianon/gosu/releases/download/1.11/gosu-amd64 -o /usr/local/bin/gosu
     chmod +x /usr/local/bin/gosu
     gosu --version
 fi
 
-# Add ddde user and group
+# Create a group and user for dde with specified IDs
 groupadd -g $DDE_GID -o dde || true
-useradd -d /home/dde -u $DDE_UID -g $DDE_GID -c "dde" -s /bin/sb -N -o -m dde -r || true
+useradd -d /home/dde -u $DDE_UID -g $DDE_GID -c "dde" -s "$SHELL_TO_INSTALL" -N -o -m dde -r || true
+
+# Install oh-my-zsh for the dde user if zsh is the selected shell
+if [ "$SHELL_TO_INSTALL" = "/bin/zsh" ] && which zsh > /dev/null 2>&1; then
```

Comment:
@sbaerlocher not everyone choosing zsh as default shell will want to use oh my zsh. I would remove this and add it into the project build if needed